src/main/java/org/broadinstitute/consent/http/db/mapper/DatasetReducer.java

```diff
@@ -75,6 +76,25 @@ && hasColumn(rowView, "property_value", String.class)) {
         }
       }
     }
+
+    if (hasColumn(rowView, "fso_file_storage_object_id", Integer.class)
+      && Objects.nonNull(rowView.getColumn("fso_file_storage_object_id", Integer.class))
+    ) {
+      FileStorageObject fileStorageObject = rowView.getRow(FileStorageObject.class);
+
+      switch (fileStorageObject.getCategory()) {
+        case NIH_INSTITUTIONAL_CERTIFICATION -> {
+          if (Objects.isNull(dataset.getNihInstitutionalCertificationFile())
+                  || fileStorageObject.getLatestUpdateDate().isAfter(dataset.getNihInstitutionalCertificationFile().getLatestUpdateDate())) {
```

Comment:
Can dataset.getNihInstitutionalCertificationFile().getLatestUpdateDate() return null in this block?

Reply:
Since there is a non-null constraint in the database for a create date, there should never be a null case here.

Reply:
Fair enough. I tend to be on the side of expecting those values (null) and defending against them.  YMMV.